chia/consensus/block_body_validation.py

```diff
@@ -103,18 +105,34 @@ def include_reward_coins(self, block: FullBlock) -> None:
             timestamp = block.foliage_transaction_block.timestamp
             coin_id = coin.name()
             assert coin_id not in self.additions_since_fork
-            self.additions_since_fork[coin_id] = ForkAdd(coin, block.height, timestamp, None, True)
+            self.additions_since_fork[coin_id] = ForkAdd(coin, block.height, timestamp, None, True, False)
 
     def include_spends(self, conds: Optional[SpendBundleConditions], block: FullBlock, header_hash: bytes32) -> None:
         self.update_fork_peak(block, header_hash)
         if conds is not None:
             assert block.foliage_transaction_block is not None
             timestamp = block.foliage_transaction_block.timestamp
             for spend in conds.spends:
-                self.removals_since_fork[bytes32(spend.coin_id)] = ForkRem(bytes32(spend.puzzle_hash), block.height)
+                spend_coin_id = bytes32(spend.coin_id)
+                spend_ph = bytes32(spend.puzzle_hash)
+                self.removals_since_fork[spend_coin_id] = ForkRem(spend_ph, block.height)
+                # We're only relying on this flag here to tells us that the
+                # coin has the same puzzle hash and amount as its parent and
+                # the amount is odd (among other things).
+                spend_same_as_parent = bool(spend.flags & ELIGIBLE_FOR_FF)
```

Comment:
This flag means more things than just same puzzle hash, same amount and odd amount.
I don't understand why this should be considered at all. This is setting the `spent_index` differently than it's set during a reorg, isn't it?
Imagine coming here and `ELIGIBLE_FOR_FF` is not set, but puzzle hash and amount do match. You would still set `same_as_parent` to `False`. If this coin is later spent, and then reorged back into being unspent, you will set this flag *only* based on whether the puzzle hasn and amount matches. That's my understanding at least. It would make this field set inconsistently. I think that adds unnecessary complexity.

Am I misunderstanding this?

Reply:
Originally the goal was to try to get as closely as possible to an actual fast forward singleton, but as for instance the reorg case loses information, we're now just dealing with simple same-as-parent coins where we'll accept any coins that match parent puzzle hash and amount, and this flag isn't considered anymore.